runtime/src/message_processor.rs

```diff
@@ -502,6 +504,14 @@ impl MessageProcessor {
             let mut time = Measure::start("execute_instruction");
             let pre_remaining_units = compute_meter.borrow().get_remaining();
 
+            let program_id = instruction.program_id(&message.account_keys);
+            if invoke_context.is_feature_active(&prevent_calling_precompiles_as_programs::id())
+                && is_precompile(program_id, |id| invoke_context.is_feature_active(id))
+            {
+                // Precompiled programs don't have an instruction processor
+                return Ok(());
```

Comment:
hmm, maybe this should be `continue`? (at this moment, adding a failing unit test would be desirable...)

Is it ok to bail out early from `message`'s instruction processing? I mean, following instructions should be executed. and they could fail. in that case, we should revert all side-effects by preceding instruction so far?

Reply:
Good catch, that was a merge error, that block used to be in a function that just processed an instruction.  I'l add a test to catch it too.